session.go

```diff
@@ -199,6 +223,32 @@ func SetKeyValueDeletedCallback(f KeyValueDeletedCallback) {
 	keyValueDeletedCallback = f
 }
 
+// SetBulkDataResponseReceiver register a callback function when it received request from a DataStream.
+func SetBulkDataResponseReceiver(streamID string, receiver BulkDataResponseReceiver) error {
+	bulkDataResponseReceivers[strings.ToLower(streamID)] = receiver
+
+	ctrl := &sessCtrlSubscribeBulkDataResponse{
+		subscribe: &DeviceSubscribeBulkDataResponse{StreamID: streamID},
+		response:  make(chan error, 1),
+	}
+
+	sessCtrl.subscribeBulkDataResponse <- ctrl
+	return <-ctrl.response
+}
+
+func resubscribeBulkDataResponse() error {
+	logInfo("[Session] Start re-setting SubscribeBulkDataResponse")
+	defer logInfo("[Session] Finished re-setting SubscribeBulkDataResponse")
+
+	for streamID, receiver := range bulkDataResponseReceivers {
+		if err := SetBulkDataResponseReceiver(streamID, receiver); err != nil {
```

Comment:
Is this iterating through a map, and then calling a function that sets the same value in the map in [L228](https://github.com/moderepo/device-sdk-go/blob/a293e76edf94ee995fabfa89562a23d6fd5228a7/session.go#L228)? I guess Golang is okay with it, but it seems bad to insert into a map while you are iterating on the map. If it's called from somewhere else, SetBulkDataResponseReceiver should check before inserting.

Reply:
Oh, nice catch. fixed: https://github.com/moderepo/device-sdk-go/pull/28/commits/4fbd68a6119b57bc6d3087756493687598cf7342